lib/literature/uploader.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Literature.Uploader do
+  @moduledoc """
+    Literature Uploader
+  """
+  use Waffle.Definition
+  use Waffle.Ecto.Definition
+
+  alias Literature.Config
+
+  @acl :public_read
+  @extension_whitelist ~w(.jpg .jpeg .png)
+  @versions ~w(w100 w200 w300 w400 w500 w600 w700 w800 w900 original)a
+
+  def asset_host, do: Config.waffle_asset_host()
+  def bucket, do: Config.waffle_bucket()
+  def __storage, do: Config.waffle_storage()
+
+  # Whitelist file extensions:
+  def validate({file, _}) do
+    Enum.member?(
+      @extension_whitelist,
+      file.file_name
+      |> Path.extname()
+      |> String.downcase()
+    )
+  end
+
+  # Define w100 transformation:
+  def transform(:w100, _) do
+    {:convert, "-resize 10% -format webp", :webp}
```

Comment:
please verify this makes the max width 100px and not 10% of the size of the original file